src/pdfprocessor/pdf_processor.py

```diff
@@ -0,0 +1,107 @@
+"""PDF Merger app."""
+
+import os
+import shutil
+
+from pypdf import PdfWriter, PdfReader
+from simian.gui import Form, component, utils
+
+
+def gui_init(meta_data: dict) -> dict:
+    # Create the form and load the json builder into it.
+    Form.componentInitializer(description=extend_description(mode=meta_data["mode"]))
+    form = Form(from_file=__file__)
+
+    return {
+        "form": form,
+        "navbar": {"title": "PDF Merger", "subtitle": "<small>Simian demo</small>"},
+    }
+
+
+def gui_event(meta_data: dict, payload: dict) -> dict:
+    Form.eventHandler(MergeFiles=merge_files)
+    callback = utils.getEventFunction(meta_data, payload)
+    return callback(meta_data, payload)
+
+
+def gui_close(meta_data):
+    # Remove the session folder on app close.
+    shutil.rmtree(utils.getSessionFolder(meta_data), ignore_errors=True)
+
+
+def extend_description(mode):
+    """Extend the description based on the run mode."""
+
+    def inner(comp):
+        if mode == "deployed":
+            comp.content += "<li>Warning: do not upload sensitive information.</li><li>Tip: click the 'Close' button in the top-right corner to close the app AND immediately delete the uploaded and merged files. Remaining files from your session are deleted every 24 hours.</li>"
+        else:
+            comp.content += "<li>Tip: selected and merged files are deleted when you close the app."
```

Comment:
Although this may be true for the artifacts in the session data, I think we should make clear that the source files and the result saved elsewhere on the system are not touched.